### PR TITLE
Add export and import for profiles

### PR DESCRIPTION
**Pull request by dhaavi**

### CODE REVIEW

File: sync/profile.go [hunk 1]
```diff
@@ -1,45 +1,440 @@
...
+	Internal   bool       `json:"internal,omitempty"`
+}
+
+// ProfileIcon represents a profile icon.
+type ProfileIcon struct {
+	Type  profile.IconType `json:"type"`
+	Value string           `json:"value"`
+}
+
+// ProfileIcon represents a profile fingerprint.
```
Comment (ppacher) [inline sync/profile.go:51]: Comment is for `ProfileIcon` not for `ProfileFingerprint`

File: sync/profile.go [hunk 2]
```diff
@@ -1,45 +1,440 @@
...
+				ValidateOnly: q.Has("validate"),
+				RawExport:    string(ar.InputData),
+				RawMime:      ar.Header.Get("Content-Type"),
+			},
+			AllowUnknown: q.Has("allowUnknown"),
+			AllowReplace: q.Has("allowReplace"),
+		}
+	} else {
+		request = &ProfileImportRequest{}
+		if err := json.Unmarshal(ar.InputData, request); err != nil {
```
Comment (ppacher) [inline sync/profile.go:184]: Allocating an empty struct for `request` is not required, json.Unmarshal will do it anyway

File: sync/profile.go [hunk 3]
```diff
@@ -1,45 +1,440 @@
...
+func ExportProfile(scopedID string) (*ProfileExport, error) {
+	// Get Profile.
+	r, err := db.Get(profile.ProfilesDBPath + scopedID)
+	if err != nil {
+		return nil, fmt.Errorf("%w: failed to find profile: %w", ErrTargetNotFound, err)
+	}
+	p, err := profile.EnsureProfile(r)
+	if err != nil {
+		return nil, fmt.Errorf("%w: failed to load profile: %w", ErrExportFailed, err)
+	}
```
Comment (ppacher) [inline sync/profile.go:220]: Shouldn't the profile get locked here?

File: sync/settings.go [hunk 4]
```diff
@@ -334,3 +298,48 @@ func ImportSettings(r *SettingsImportRequest) (*ImportResult, error) {
...
+
+	// Validate config and gather some metadata.
+	var checked int
+	err = config.ForEachOption(func(option *config.Option) error {
+		// Check if any setting is set.
+		// TODO: Fix this - it only checks for global settings.
+		// if r.Reset && option.IsSetByUser() {
+		// 	result.ReplacesExisting = true
+		// }
+
```
Comment (ppacher) [inline sync/settings.go:313]: Is this something that needs to be sorted out before we release the import/export feature?

**Review by ppacher — CHANGES_REQUESTED** [review]

File: sync/profile.go (hunk 2 above)
Comment (dhaavi) [reply sync/profile.go:184]: > Unmarshal parses the JSON-encoded data and stores the result in the value pointed to by v. If v is nil or not a pointer, Unmarshal returns an InvalidUnmarshalError.

File: sync/profile.go (hunk 3 above)
Comment (dhaavi) [reply sync/profile.go:220]: This is parsing a raw entry from the DB. Nobody has can have a lock on this.

**Pull request merged**